### PR TITLE
Plugin Creator automatically reloads when running and ALT+R activates…

### PR DESCRIPTION
**Pull request by worldwideweary**
Resolves: #154 … Run

Testing to see a pull request against your repository. Check it out see if you think it could be useful. Works decent for me.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Can't you include the `save` ?

File: mscore/plugin/pluginCreator.ui [hunk 1]
```diff
@@ -46,7 +46,7 @@
           </sizepolicy>
          </property>
          <property name="text">
-          <string>Run</string>
+          <string>&amp;Run</string>
```
Comment (Jojo-Schmitz) [inline mscore/plugin/pluginCreator.ui:49]: While a good idea in general, it'd render all translation invalid, and without a proper tranbslation infrastructure in place

File: mscore/plugin/pluginCreator.ui [hunk 2]
```diff
@@ -59,7 +59,7 @@
           </sizepolicy>
          </property>
          <property name="text">
-          <string>Stop</string>
+          <string>&amp;Stop</string>
```
Comment (Jojo-Schmitz) [inline mscore/plugin/pluginCreator.ui:62]: dito

File: mscore/plugin/pluginCreator.ui (hunk 1 above)
Comment (worldwideweary) [reply mscore/plugin/pluginCreator.ui:49]: Ah, makes sense. Well, at least the idea is here for anyone concerned

**Comment by worldwideweary** [comment]
...with your suggestions

**Comment by worldwideweary** [comment]
Adding 'Run' and 'Stop' to the file menu would be a way to manifest keyboard shortcut associations. Would that a problem for translations if they're spelled exactly the same as the buttons?

File: mscore/plugin/pluginCreator.ui [hunk 3]
```diff
@@ -89,7 +89,7 @@
      <x>0</x>
      <y>0</y>
      <width>800</width>
-     <height>21</height>
+     <height>20</height>
```
Comment (worldwideweary) [inline mscore/plugin/pluginCreator.ui:92]: Qt Creator is funny sometimes

Comment (Jojo-Schmitz) [reply mscore/plugin/pluginCreator.ui:92]: Yep. Just don't commit that

Comment (worldwideweary) [reply mscore/plugin/pluginCreator.ui:92]: yeah, on command line in *nix I don't mess with trying to omit per-line. Reminds me of some stuff back in 3.4-3.6 days :) At any rate, thanks for the recommendation to alternatively save when editor has changes.

**Comment by Jojo-Schmitz** [comment]
Plugins > Plugin creator > Run works, but F5 does not, not even when Plugin creator is the top and active window
Same for Stop and F7.

**Comment by worldwideweary** [comment]
> Plugins > Plugin creator > Run works, but F5 does not, not even when Plugin creator is the top and active window Same for Stop and F7.

Shoot. After testing looks like I get "Ambiguous shortcut overload: F5". Wonder what constitutes its ambiguity? Any suggestions for a shortcut? Maybe Ctrl+R and Ctrl+X... will see.

File: mscore/plugin/pluginCreator.cpp [hunk 4]
```diff
@@ -67,6 +67,9 @@ PluginCreator::PluginCreator(QWidget* parent)
 
       actionQuit->setShortcut(QKeySequence(QKeySequence::Close));
 
+      actionRun->setShortcut(QKeySequence(QKeySequence::fromString("F5")));
+      actionStop->setShortcut(QKeySequence(QKeySequence::fromString("F7")));
```
Comment (Jojo-Schmitz) [inline mscore/plugin/pluginCreator.cpp:71]: Why not just
```cpp
      actionRun->setShortcut(QKeySequence("F5"));
      actionStop->setShortcut(QKeySequence("F7"));
```

**Comment by Jojo-Schmitz** [comment]
Maybe it needs to be in shortcuts.xml?

File: mscore/plugin/pluginCreator.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply mscore/plugin/pluginCreator.cpp]: Hmm, not the shortcut hints don't show in the menu

File: mscore/plugin/pluginCreator.cpp [hunk 5]
```diff
@@ -279,6 +281,9 @@ static void qmlMsgHandler(QtMsgType type, const QMessageLogContext &, const QStr
 
 void PluginCreator::runClicked()
       {
+      if (state == PCState::DIRTY)
+            savePlugin();
+      else load();
```
Comment (Jojo-Schmitz) [inline mscore/plugin/pluginCreator.cpp:286]: Can you fix this formatting issue to 2 lines and indentation, please

File: mscore/plugin/pluginCreator.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply mscore/plugin/pluginCreator.cpp]: New PR?

**Pull request closed without merging**